### PR TITLE
add nvidia info to launch docker setup

### PR DESCRIPTION
**Pull request by bcsherma**
## Description

In the launch docs overhaul we accidentally lost the section dedicated to setting up the nvidia container toolkit.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [ ] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [ ] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Review by zacharyblasczyk — APPROVED** [review]

File: docs/guides/launch/setup-launch-docker.md [hunk 1]
```diff
@@ -1,16 +1,18 @@
 ---
 displayed_sidebar: default
 ---
+
```
Comment (KyleGoyette) [inline docs/guides/launch/setup-launch-docker.md:4]: errant new line?

File: docs/guides/launch/setup-launch-docker.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
 # Set up for Docker
+
```
Comment (KyleGoyette) [inline docs/guides/launch/setup-launch-docker.md:9]: errant new line?

File: docs/guides/launch/setup-launch-docker.md [hunk 3]
```diff
@@ -71,26 +72,52 @@ Create a queue that uses Docker as compute resource with the W&B CLI:
 6. Define your Docker queue configuration in the **Configuration** field.
 7. Click on the **Create Queue** button to create the queue.
 
-
-## Configure a launch agent on a local machine 
+## Configure a launch agent on a local machine
```
Comment (KyleGoyette) [inline docs/guides/launch/setup-launch-docker.md:95]: errant deleted line?

File: docs/guides/launch/setup-launch-docker.md [hunk 4]
```diff
@@ -32,14 +34,11 @@ There are two syntax transformations that take place:
...
 {
-    "env": [
-        "MY_ENV_VAR=value",
-        "MY_EXISTING_ENV_VAR"
-    ],
-    "volume": "/mnt/datasets:/mnt/datasets",
-		"rm": true
+  "env": ["MY_ENV_VAR=value", "MY_EXISTING_ENV_VAR"],
+  "volume": "/mnt/datasets:/mnt/datasets",
+  "rm": true
```
Comment (KyleGoyette) [inline docs/guides/launch/setup-launch-docker.md]: ahk missed this in my spacing check

File: docs/guides/launch/setup-launch-docker.md [hunk 5]
```diff
@@ -71,26 +72,52 @@ Create a queue that uses Docker as compute resource with the W&B CLI:
...
+
+The `--gpus` flag of the `docker run` command allows you to specify which GPU are available to a container.
+Use the `gpus` key in the launch queue configuration to specify the value that will be passed to `--gpus` when the agent runs your job on Docker. See [configure a Docker queue](#configure-a-docker-queue) for more information on customizing `docker run` argument set by the agent.
+
+:::tip
+To have each run use all available GPU, add:
+
+```json
+{
+  "gpus": "all"
```
Comment (KyleGoyette) [inline docs/guides/launch/setup-launch-docker.md]: We should really add the accelerator image here as well :( . A user today just immediately ran into this.

**Review by KyleGoyette — COMMENTED** [review]
Overall looks good, I'm wondering if some formatting was autoapplied though

File: docs/guides/launch/setup-launch-docker.md [hunk 6]
```diff
@@ -71,26 +72,52 @@ Create a queue that uses Docker as compute resource with the W&B CLI:
...
-## Configure a launch agent on a local machine 
+## Configure a launch agent on a local machine
 
 Configure the launch agent with a YAML config file named `launch-config.yaml`. By default, W&B will check for the config file in `~/.config/wandb/launch-config.yaml`. You can optionally specify a different directory when you activate the launch agent.
 
 :::tip
 You can use the W&B CLI to specify core configurable options for the launch agent (instead of the config YAML file): maximum number of jobs, W&B entity, and launch queues. See the [`wandb launch-agent`](../../ref/cli/wandb-launch-agent.md) command for more information.
 :::
 
+### Nvidia GPU
```
Comment (ngrayluna) [inline docs/guides/launch/setup-launch-docker.md]: Let's make the title an action. E.g. "Enable launch agent to use GPU" (or something)

File: docs/guides/launch/setup-launch-docker.md [hunk 7]
```diff
@@ -71,26 +72,52 @@ Create a queue that uses Docker as compute resource with the W&B CLI:
...
+In order to use GPU from within a Docker container, you must install the [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker). To test your toolkit install, try running:
+
+```bash
+docker run --gpus all nvidia/cuda:11.0-base nvidia-smi
+```
+
+If you see the output from `nvidia-smi`, then your toolkit should be installed correctly!
+
+The `--gpus` flag of the `docker run` command allows you to specify which GPU are available to a container.
+Use the `gpus` key in the launch queue configuration to specify the value that will be passed to `--gpus` when the agent runs your job on Docker. See [configure a Docker queue](#configure-a-docker-queue) for more information on customizing `docker run` argument set by the agent.
```
Comment (ngrayluna) [inline docs/guides/launch/setup-launch-docker.md]: Debating if this content should be moved up into the launch queue config section? since it is in relation to the launch queue's target environment?

Or rather, is there a way to specify to use GPUs in a launch agent's config? 

TL;DR let's make it clear if using GPUs is a part of a launch queue's target environment config vs. setting up an agent to be allowed to execute jobs on a local machine?   i.e. Let's make it clear the logic of what and where to specify parameters (launch queue vs launch agent config).

**Review by ngrayluna — CHANGES_REQUESTED** [review]

File: docs/guides/launch/setup-launch-docker.md (hunk 7 above)
Comment (bcsherma) [reply docs/guides/launch/setup-launch-docker.md]: Yeah there are really two steps to enable GPU:
- make sure you are setting `--gpus` in your queue config
- make sure you have the nvidia container toolkit setup properly

I would propose the following compromise: include the `--gpus` argument in the example docker queue config where we demonstrate how the config is mapped to docker run arguments.

File: docs/guides/launch/setup-launch-docker.md (hunk 1 above)
Comment (bcsherma) [reply docs/guides/launch/setup-launch-docker.md:4]: this is done automatically by prettier in my vscode. I wouldn't say errant, just motivated by style.

File: docs/guides/launch/setup-launch-docker.md (hunk 5 above)
Comment (bcsherma) [reply docs/guides/launch/setup-launch-docker.md]: I've added in just below in the same tip

File: docs/guides/launch/setup-launch-docker.md (hunk 2 above)
Comment (bcsherma) [reply docs/guides/launch/setup-launch-docker.md:9]: prettier

File: docs/guides/launch/setup-launch-docker.md (hunk 3 above)
Comment (bcsherma) [reply docs/guides/launch/setup-launch-docker.md:95]: prettier

**Review by KyleGoyette — APPROVED** [review]

**Comment by ngrayluna** [comment]
> Yeah there are really two steps to enable GPU:
> 
> * make sure you are setting `--gpus` in your queue config
> * make sure you have the nvidia container toolkit setup properly
> 
> I would propose the following compromise: include the `--gpus` argument in the example docker queue config where we demonstrate how the config is mapped to docker run arguments.



@bcsherma @KyleGoyette Incorporated said changes to queue section of this page. Let me know if this works. Preview of how this looks included int his PDF
[Set up for Docker _ Weights & Biases Documentation.pdf](https://github.com/wandb/docodile/files/13271421/Set.up.for.Docker._.Weights.Biases.Documentation.pdf)

**Review by KyleGoyette — APPROVED** [review]

**Review by ngrayluna — APPROVED** [review]
